File: leecode/code/3/3.java
class Solution {

    public int lengthOfLongestSubstring(String s) {
        int left = 0;
        int right = 0;
        int max = 0;

        while (right < s.length() && left <= right) {
            String temp = "";
            left = right + 1;
            while (right < s.length()) {
                if (temp.indexOf(s.charAt(right)) == -1) {
                    temp = temp + s.charAt(right);
                    right++;
                } else {
                    break;
                }
            }
            max = Math.max(max, temp.length());
            right = left;
        }

        return max;
    }
}

File: leecode/code/3/3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        res = 0;
        left = 0;
        right = 0;
        
        while left <= right and right < len(s) :
            record = set()
            while right < len(s) and s[right] not in record :
                record.add(s[right])
                right += 1
            
            res = max(len(record), res)
            left += 1
            right = left     
        return res